src/main/java/seedu/address/logic/commands/delete/DeleteCommandExecutor.java

```diff
@@ -0,0 +1,38 @@
+package seedu.address.logic.commands.delete;
+
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+
+/**
+ * Contains the logic for executing the delete command.
+ */
+public class DeleteCommandExecutor {
+
+    /**
+     * Deletes a client from the address book using their index in the last shown list.
+     */
+    public static Person deleteByIndex(List<Person> lastShownList, Index targetIndex) throws CommandException {
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
```

Comment:
```suggestion
        if (targetIndex.getZeroBased() >= lastShownList.size() || targetIndex.getZeroBased() < 1) {
```

Reply:
Perhaps you can consider adding this extra check, which also includes the case where 0 or negative numbers are passed in, which is invalid as well.
Similar for EditCommand line 85